pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -299,47 +313,12 @@ func (vs *VSphere) getVMByName(ctx context.Context, nodeName k8stypes.NodeName)
 // NodeAddresses is an implementation of Instances.NodeAddresses.
 func (vs *VSphere) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress, error) {
 	// Get local IP addresses if node is local node
-	if vs.localInstanceID == nodeNameToVMName(nodeName) {
+	if vs.hostName == nodeNameToVMName(nodeName) {
 		return getLocalIP()
 	}
-	addrs := []v1.NodeAddress{}
```

Comment:
Objective of this task is to report hostName and not VMName.
Since VMName is not needed to report there is no need to connect to VC and generate VMName